__tests__/app.test.js

```diff
@@ -154,6 +154,59 @@ describe("GET /api/articles/:article_id/comments", () => {
   });
 });
 
+describe("POST /api/articles/:article_id/comments", () => {
+  test("Should respond with 201 and respond with the posted comment after being added to db", () => {
+    const newComment = {
+      author: "icellusedkars",
+      body: "This comment sure seems pointless!",
+    };
+    return request(app)
+      .post("/api/articles/3/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.comment).toMatchObject({
+          comment_id: expect.any(Number),
+          body: expect.any(String),
+          article_id: expect.any(Number),
+          author: expect.any(String),
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+        });
+      });
+  });
+  test("Should respond with 400 Bad Request when article_id is an invalid type when trying to post comment", () => {
+    const newComment = {
+      author: "icellusedkars",
+      body: "This comment sure seems pointless!",
+    };
+    return request(app)
+      .post("/api/articles/notanumber/comments")
+      .send(newComment)
+      .expect(400);
+  });
+  test("Should respond with 400 Bad Request when posting a comment with incorrect author type", () => {
+    const newCommentBadAuthor = {
+      author: 2,
+      body: "This comment sure seems pointless!",
+    };
+    return request(app)
+      .post("/api/articles/3/comments")
+      .send(newCommentBadAuthor)
+      .expect(400);
+  });
+  test("Should respond with 400 Bad Request when posting a comment with non-existent author", () => {
+    const newCommentBadAuthor = {
+      author: "joe",
+      body: "This comment sure seems pointless!",
+    };
+    return request(app)
+      .post("/api/articles/3/comments")
+      .send(newCommentBadAuthor)
+      .expect(400);
+  });
```

Comment:
Rethink the status code for this request - the user isn't requesting 'badly' they're just looking for a user that isn't in our database. Similar to when a article id is valid but just isn't found.

I'd also expect a msg with this response.